File: C#/8-kyu/Exclamation-marks-series-#4-Remove-all-exclamation-marks-from-sentence-but-ensure-a-exclamation-mark-at-the-end-of-string.cs
// Description:
// Remove all exclamation marks from sentence but ensure a exclamation mark at the end of string. For a beginner kata, you can assume that the input data is always a non empty string, no need to verify it.

// Examples
// remove("Hi!") === "Hi!"
// remove("Hi!!!") === "Hi!"
// remove("!Hi") === "Hi!"
// remove("!Hi!") === "Hi!"
// remove("Hi! Hi!") === "Hi Hi!"
// remove("Hi") === "Hi!"

public class Kata
{
  public static string Remove(string s)
  {
    return s.Replace("!", "") + '!';
  }
}

namespace Solution {
  using NUnit.Framework;
  using System;

  // TODO: Replace examples and use TDD development by writing your own tests

  [TestFixture]
  public class Test
  {
    [Test]
    public void SampleTest()
    {
      Assert.AreEqual("Hi!", Kata.Remove("Hi!"));
      Assert.AreEqual("Hi!", Kata.Remove("Hi!!!"));
      Assert.AreEqual("Hi!", Kata.Remove("!Hi"));
      Assert.AreEqual("Hi!", Kata.Remove("!Hi!"));
      Assert.AreEqual("Hi Hi!", Kata.Remove("Hi! Hi!"));
      Assert.AreEqual("Hi!", Kata.Remove("Hi"));
    }
  }
}

File: JavaScript/8-kyu/Exclamation-marks-series-#4-Remove-all-exclamation-marks-from-sentence-but-ensure-a-exclamation-mark-at-the-end-of-string.js
// Description:
// Remove all exclamation marks from sentence but ensure a exclamation mark at the end of string.For a beginner kata, you can assume that the input data is always a non empty string, no need to verify it.

//     Examples
// remove("Hi!") === "Hi!"
// remove("Hi!!!") === "Hi!"
// remove("!Hi") === "Hi!"
// remove("!Hi!") === "Hi!"
// remove("Hi! Hi!") === "Hi Hi!"
// remove("Hi") === "Hi!"

function remove(s) {
  let removeBangs = s.replace(/!/g, "");
  return removeBangs.concat("", "!");
}

describe("Basic Tests", function() {
  it("It should works for basic tests", function() {
    Test.assertSimilar(remove("Hi!"), "Hi!");
    Test.assertSimilar(remove("Hi!!!"), "Hi!");
    Test.assertSimilar(remove("!Hi"), "Hi!");
    Test.assertSimilar(remove("!Hi!"), "Hi!");
    Test.assertSimilar(remove("Hi! Hi!"), "Hi Hi!");
    Test.assertSimilar(remove("Hi"), "Hi!");
  });
});
